src/overlays/actors/ovl_Bg_Spot18_Obj/z_bg_spot18_obj.c

```diff
@@ -21,47 +41,259 @@ const ActorInit Bg_Spot18_Obj_InitVars = {
+    if (this->dyna.unk_150 < -0.001f) {
+        if (this->unk_168 <= 0) {
+            func_808B8EE0(this);
+        }
+    } else {
+        this->unk_168 = 20;
+    }
+    func_808B8E20(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8E20.s")
+void func_808B8EE0(BgSpot18Obj* this) {
+    this->actionFunc = func_808B8F08;
+    this->dyna.actor.posRot.rot.y = 0;
+    this->dyna.actor.speedXZ = 0.0f;
+    this->dyna.actor.velocity.z = 0.0f;
+    this->dyna.actor.velocity.y = 0.0f;
+    this->dyna.actor.velocity.x = 0.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8E64.s")
+void func_808B8F08(BgSpot18Obj* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8E7C.s")
+    Math_ApproxF(&this->dyna.actor.speedXZ, 1.2f, 0.1f);
+    Actor_MoveForward(&this->dyna.actor);
+    func_808B8DDC(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8EE0.s")
+    if (func_800CB594(this->dyna.actor.posRot.pos.x, this->dyna.actor.posRot.pos.z, this->dyna.actor.initPosRot.pos.x,
+                      this->dyna.actor.initPosRot.pos.z) >= 6400.0f) {
+        func_808B9030(this);
+        this->dyna.actor.posRot.pos.x =
+            (Math_Sins(this->dyna.actor.posRot.rot.y) * 80.0f) + this->dyna.actor.initPosRot.pos.x;
+        this->dyna.actor.posRot.pos.z =
+            (Math_Coss(this->dyna.actor.posRot.rot.y) * 80.0f) + this->dyna.actor.initPosRot.pos.z;
+        this->dyna.unk_150 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F);
+        func_80078884(0x4802);
+        Audio_PlayActorSound2(&this->dyna.actor, 0x2835);
+    } else {
+        func_8002F974(&this->dyna.actor, 0x200A);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B8F08.s")
+void func_808B9030(BgSpot18Obj* this) {
+    this->actionFunc = func_808B9040;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B9030.s")
+void func_808B9040(BgSpot18Obj* this, GlobalContext* globalCtx) {
+    func_808B8E20(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/func_808B9040.s")
+void BgSpot18Obj_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Obj* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/BgSpot18Obj_Update.s")
+    if (this->unk_168 > 0) {
+        this->unk_168 -= 1;
+    }
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Obj/BgSpot18Obj_Draw.s")
+void BgSpot18Obj_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Gfx_DrawDListOpa(globalCtx, sDlists[THIS->dyna.actor.params & 0xF]);
```

Comment:
Note sure if thisx works here but if it does it's fine to use it directly imo. Otherwise I would only inline `THIS` if it's the only way (ie. a `this` temp doesn't work).
```suggestion
    Gfx_DrawDListOpa(globalCtx, sDlists[thisx->params & 0xF]);
```

Reply:
Yeah, a temp doesn't work so I'll change it to thisx